### DISCUSSION TITLE
Support for 12hour Format

### DISCUSSION THREAD

**Discussion by nextecsolutions** [Makuna/Rtc#204] [Q&A] [answered]
Hi,
 I am struggling to get the time in 12 hour format, Can you please help to determine what changes i have to make in the code so that i can get my clock to work in 12hr format ,Thanks.

**Comment by Makuna** [comment]
This library internally only uses 24 hour time.  The display of that time is left to the sketch writer.  You need to convert it before displaying it. 
If you have an input model that has key input, then you may also need conversion from 12 hour to 24 hour before setting an RtcDateTime object.
If you have an input model that just increments or decrements, you apply that to the 24 hour and rely on your display conversion to update the active editing display; thus no need for 12 hour to 24 hour.
If you are looking for the best way to actually code this, use your favorite search engine and look for "C++ converting 24 time to 12 hour time".

**Comment by Makuna** [comment]
If you can wait, I do plan to port some code I use in my own projects into the library for converting 24hour to 12 hour and back.

**Reply by Makuna** [reply, marked as answer]
Checked in.

To convert a 24 hour to 12 hour with am and pm, you do this...
```
RtcHourAmPm hourAmPm = dt.Hour();

or

RtcHourAmPm hourAmPm = dt.HourAmPm();
```

then the hourAmPm object contains accessors for hour in 12 hour format and meridiem for am or pm.

```
Serial.print(hourAmPm.Hour());
if (hourAmPm.Meridiem() == Rtc_AM)
{
    Serial.print("am");
}
else
{
    Serial.print("pm");
}
```

**Comment by nextecsolutions** [comment]
Thankyou Sir, I will try this and let you know.

Regards,
Bilal
